Avoid the race between checking a remote state and the decision.

--- src/virtualbox7/src/VBox/VMM/VMMR3/VMR3Emt.cpp
+++ src/virtualbox7/src/VBox/VMM/VMMR3/VMR3Emt.cpp
@@ -922,9 +922,10 @@ static DECLCALLBACK(void) vmR3HaltGlobal1NotifyCpuFF(PUVMCPU pUVCpu, uint32_t fF
                     AssertRC(rc);
                 }
             }
-            else if (   enmState == VMCPUSTATE_STARTED_EXEC_NEM
-                     || enmState == VMCPUSTATE_STARTED_EXEC_NEM_WAIT)
+            else {// if (   enmState == VMCPUSTATE_STARTED_EXEC_NEM
+                  //   || enmState == VMCPUSTATE_STARTED_EXEC_NEM_WAIT)
                 NEMR3NotifyFF(pUVCpu->pVM, pVCpu, fFlags);
+            }
         }
     }
     /* This probably makes little sense: */
@@ -1064,8 +1065,8 @@ static DECLCALLBACK(void) vmR3DefaultNotifyCpuFF(PUVMCPU pUVCpu, uint32_t fFlags
         if (pVCpu)
         {
             VMCPUSTATE enmState = pVCpu->enmState;
-            if (   enmState == VMCPUSTATE_STARTED_EXEC_NEM
-                || enmState == VMCPUSTATE_STARTED_EXEC_NEM_WAIT)
+//            if (   enmState == VMCPUSTATE_STARTED_EXEC_NEM
+//                || enmState == VMCPUSTATE_STARTED_EXEC_NEM_WAIT)
                 NEMR3NotifyFF(pUVCpu->pVM, pVCpu, fFlags);
         }
     }
